### PR TITLE
Refactors health doll, replaces "wounded" screen alert 

### PR DESCRIPTION
**Pull request by MrMelbert**
## About The Pull Request

The screen alert for being wounded is deleted

Instead, your health doll will now glow red on any (and all) wounded limbs

https://github.com/user-attachments/assets/83565684-3e19-4753-8034-d8de6574e2b8

To accomplish this, the doll was refactored a bit. No longer operates off of overlays, now uses vis contents across every limb, and just updates the icon state of all those vis contents

## Why It's Good For The Game

Wounds really spam the hell out of you with screen alerts and it often blocks you from seeing more relevant alerts that you care about

The mere presence of a minor dislocation on your leg prevents you from noticing that you are no longer breathing. This is a bit troublesome

This can be resolved in other ways, of course - adding a priority value to alerts? - but instead, I think we can just make better use of our existing hud elements

I find this decently intuitive, at a glance. 
- The old screen alert gave you a tooltip saying you could click the alert to examine yourself. 
- Clicking on the health doll examines yourself the exact same way.
- So, players may see their doll glowing red, and click on it to self examine, to see the report of them having a wound on their leg or chest or whatever.

## Changelog

:cl: Melbert
del: Having any wounds no longer gives you an alert in the top right
qol: Having any wounds now make the corresponding bodypart on your health doll (the lil dude on the right side of the screen) glow red. 
refactor: Refactored how the hud's health doll shows up for humans. Report any oddities
/:cl:

### CODE REVIEW

**Comment by newfren** [comment]
oh this is really nice for seeing where a wound is at a glance

**Comment by DATA-xPUNGED** [comment]
This put my dent at the health doll limb selection PR that I was going to do... eventually.....

File: code/_onclick/hud/screen_objects.dm [hunk 1]
```diff
@@ -724,6 +724,98 @@
...
+			icon_key = 7
+		else if(owner.stat == DEAD)
+			icon_key = "DEAD"
+		else if(SEND_SIGNAL(body_part, COMSIG_BODYPART_UPDATING_HEALTH_HUD, owner, overridable_key) & OVERRIDE_BODYPART_HEALTH_HUD)
+			icon_key = overridable_key[1] // thanks i hate it
+		else if(!owner.has_status_effect(/datum/status_effect/grouped/screwy_hud/fake_healthy))
+			var/damage = body_part.burn_dam + body_part.brute_dam
+			var/comparison = (body_part.max_damage / 5)
+			if(damage > (comparison * 4))
+				icon_key = 5
```
Comment (SmArtKar) [inline code/_onclick/hud/screen_objects.dm:778]: Maybe calculate damage percentage first and then change icons based off that? Also maybe define magic numbers but thats not very important (considering its icon states)

File: code/datums/wounds/_wounds.dm [hunk 2]
```diff
@@ -343,18 +342,18 @@
 
 	if(limb && !ignore_limb)
 		set_limb(null, replaced) // since we're removing limb's ref to us, we should do the same
-		// if you want to keep the ref, do it externally, there's no reason for us to remember it
+		// if you want to fkeep the ref, do it externally, there's no reason for us to remember it
```
Comment (SmArtKar) [inline code/datums/wounds/_wounds.dm:345]:
```suggestion
		// if you want to keep the ref, do it externally, there's no reason for us to remember it
```

**Review by SmArtKar — CHANGES_REQUESTED** [review]

**Review by Ghommie — APPROVED** [review]

**Pull request merged**